Import New ECLIPSE I/O Classes Into Main OPM Common Library

bska opened the pull request:
This change-set imports the facilities introduced in PR #699 into the main OPM Common library. We import the files into the new directory `opm/io/eclipse` and place the classes in namespace `Opm::ecl` as was suggested in the previous PR #755.

We discontinue the `testutil` library since it is no longer needed.

bska commented:
jenkins build this please

joakim-hove commented:
Thank you for splitting the PR

joakim-hove commented on CMakeLists.txt:
```diff
@@ -157,27 +157,51 @@ endif()
-              examples/test_util/ERst.cpp
-              examples/test_util/ESmry.cpp)
+  add_executable(compareECL
+    examples/test_util/EclFilesComparator.cpp
```
When things are shuffled around anyways - maybe these should be moved out of the `examples/` directory - they are not examples after all.

akva2 replied:
this is historical - the apps used to be in examples/. not protesting the change, just the rationale.

bska replied:
I'd be happy to move them elsewhere if we can think of a good location. I'm a little reluctant however to add another top-level directory in opm-common since we already have `examples/`, `external/`, `jenkins/`, `msim/`, `opm/`, `redhat/`, `src/`, and `tests/`.

joakim-hove commented on examples/test_util/EclFilesComparator.hpp:
```diff
@@ -75,7 +78,13 @@ class ECLFilesComparator {
     template <typename T>
-    void printValuesForCell(const std::string& keyword, const std::string reference, size_t kw_size, size_t cell, EGrid *grid, const T& value1, const T& value2) const;
+    void printValuesForCell(const std::string& keyword,
+                            const std::string reference,
```
Maybe the reference should be a reference :-)

akva2 commented on examples/test_util/EclRegressionTest.cpp:
```diff
@@ -812,8 +814,8 @@ void ECLRegressionTest::results_smry()
-            arrayType1.assign(nKeys, EIOD::REAL);
-            arrayType2.assign(nKeys ,EIOD::REAL);
+            arrayType1.assign(nKeys, REAL);
+            arrayType2.assign(nKeys ,REAL);
```
since you are touching anyways, fix the space please.

bska commented:
I have addressed most of the review comments but I have not yet relocated the `compareECL` and `convertECL` applications. Those still exist in `examples/test_util/`.  Feel free to suggest a better location.

bska commented:
jenkins build this please

joakim-hove commented:
> Those still exist in `examples/test_util/`. Feel free to suggest a better location.

I would put them in `test_util/` - but feel free to merge in the current form.

bska commented:
> > [Test utilities] still exist in `examples/test_util/`. Feel free to suggest a better location.
> 
> I would put them in `test_util/`.

Good idea! Addressed in revised import commit 3c29679.  I'm ready to merge this now, but I think I'll wait for #765 to reduce risk of merge conflicts.

bska commented:
jenkins build this please

bska commented:
All prerequisites for this PR have been resolved (merged into master), and Jenkins approves of the latest change-set. I now consider this to be complete. Interested parties are invited to review the current state. If there are no further comments, then I intend to merge this no later than 10am tomorrow (Thursday, 23rd of May).

bska commented:
jenkins build this please

bska commented:
jenkins build this please

bska commented:
I am going to merge this at 3pm CEST today (Thursday 23rd of May).

bska commented:
Merging this now to prepare for the next step in using ECLIPSE I/O classes from #699 as the foundation for restart I/O.
